utils/oscap-info.c

```diff
@@ -49,6 +49,27 @@
 
 #include "oscap-tool.h"
 
+// COPIED from src/XCCDF/benchmark.c
+struct xccdf_profile * xccdf_benchmark_get_profile_by_id(struct xccdf_benchmark *benchmark, const char *profile_id)
+{
+	struct xccdf_profile_iterator *profit = xccdf_benchmark_get_profiles(benchmark);
+	while (xccdf_profile_iterator_has_more(profit)) {
+		struct xccdf_profile *profile = xccdf_profile_iterator_next(profit);
+		if (profile == NULL) {
+			assert(profile != NULL);
+			continue;
+		}
+		if (strcmp(xccdf_profile_get_id(profile), profile_id)) {
```

Comment:
This is wrong. It should be `strcmp(...) == 0`. The original version in `src/XCCDF/benchmark.c` used to have `oscap_streq` here.  Due to this the function always returns the first profile from benchmark regardless of the ID :-) :-)